### ISSUE TITLE
Support for HTTPS

### ISSUE THREAD

**Issue by machineghost** [troybetz/react-youtube#85]
When I try and develop using react-youtube locally (ie. at `http://localhost:8080`) I get the following error,  intermittently:

    Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://www.youtube.com') does not match the recipient window's origin

It's referring to the `this.a.contentWindow.postMessage` line (sorry, I'm looking at the minnified version so I can't give a useful line number).  This Stack Overflow post:

http://stackoverflow.com/questions/27573017/failed-to-execute-postmessage-on-domwindow-https-www-youtube-com-http

suggest that the problem is an HTTP vs. HTTPS mismatch.

So, long story short, would it be possible for the component to look at `window.location.protocol` and use HTTP if the parent window is also on HTTP?

**Comment by tjallingt** [comment]
AFAIK: This is an error that is triggered by the YouTube Iframe API or the [`youtube-player`](https://github.com/gajus/youtube-player) package we are using to instantiate the YouTube Iframe API. This error message appears seemingly randomly and even though this error message is displayed `react-youtube` should continue to function normally.

I'm not sure where the error is triggered (especially as it is seemingly random) but since we are not setting the url of the YouTube Iframe API manually this problem would need to be solved further down the line...

**Comment by goto-bus-stop** [comment]
I also get this error on http sites with the bare YouTube iframe api. I don't think react-youtube can do anything about it. It doesn't affect the functioning of the iframe at all.

**Comment by machineghost** [comment]
Gotcha; filed https://github.com/gajus/youtube-player/issues/15 upstream :)

**Comment by tjallingt** [comment]
I'll close this issue for now; if it appears that changes to this package are needed to fix the issue it is possible to reopen it later :smile:
